### ISSUE TITLE
feature: Implement GraphQL response type

### ISSUE THREAD

**Issue by calvincestari** [apollographql/apollo-ios#3485]
This introduces a [new media type](https://graphql.github.io/graphql-over-http/draft/#sec-Media-Types) that would be used for GraphQL responses, see the [draft GraphQL over HTTP spec](https://graphql.github.io/graphql-over-http/draft/).

It looks like the work should generally be simply:
1. Modify the `accept` header to include the new media type signalling to the server that the client can accept responses of that type.
2. Add `application/graphql-response+json` to any `content-type` validation we do before parsing a server response.
3. _Errors - TBD_

**Comment by calvincestari** [comment]
@martinbonnin - I think you said something earlier about taking action on errors when the response is of the new type, is that correct?

**Comment by martinbonnin** [comment]
@calvincestari correct, depends what your behaviour is right now but typically you should throw an error if HTTP code != 200. With the advent of `graphql-response+json`, these responses can be parsed and exposed to the user instead of throwing.
